tests/test_parameters.py

```diff
@@ -57,35 +81,40 @@ def setUp(self):
         self.pB.define("shared", 2)
         self.keyB = "B"
 
-        inp = {self.keyA: self.pA, self.keyB: self.pB}
-        self.l = JointParameterList(inp, shared=["shared"])
+        self.l = JointLatent()
+        self.l.add_model_parameters(self.pA, self.keyA)
```

Comment:
Maybe it would help here to document a little bit what is happening, in particular when user take the tests to use this as a model for their implementation. Primarily what the difference between add_model_parameters and add is doing - some text documentation above the relevant lines might be sufficient.